src/Form/ConfigurationForm.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+namespace Drupal\elastic_apm\Form;
+
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Configuration form for the Elastic APM.
+ *
+ * @package Drupal\elastic_apm\Form
+ */
+class ConfigurationForm extends ConfigFormBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'elastic_apm_configuration';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getEditableConfigNames() {
+    return ['elastic_apm.configuration'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $config = $this->config('elastic_apm.configuration');
+
+    $form['application_configuration'] = [
+      '#type' => 'details',
+      '#title' => $this->t('Application configuration'),
+      '#open' => TRUE,
+    ];
```

Comment:
**CHANGE REQUEST**

Not sure that we need a parent container that hold the whole form; the point of the containers is to break the form into sub-sections, the form itself (especially when displayed on its own page) is the parent container.

Also, no need to use such long name as `application_configuration` you can use something shorter as `configuration`, even though nothing wrong if it doesn't make really long lines. Which it will, when we make the following changes.

We are defining global module configuration. There's already a lot of settings here, and we may be adding more as mentioned elsewhere. We should be grouping related settings together, here's my current suggestions:

* application
* connection
* errors

You can leave the status on its own. Group the rest first on the config schema and then on the form. Later we may add more like privacy, tags etc.

Reply:
I've added the groupings in the form, but I'd like to keep the schema as is w/o prefixing the group name like application.server_url etc., just because of the fact that I can send it straight to the PHP Agent w/o having to do any manipulation of the config array.